### ISSUE TITLE
Needless log entries to syslog for GET requests

### ISSUE THREAD

**Issue by colinmollenhour** [moby/moby#32046]
**Description**

I'm using Kontena and OSSEC so this issue became very visible due to OSSEC not recognizing these log entries and Kontena creating a log of them, but the gist is that I propose the loglevel of these entries be reduced to "warning" or lower since they are harmless but may be abundant in some cases.

Example:

```
Mar 23 10:41:14 admin1 dockerd[15092]: time="2017-03-23T10:41:14.722146447-05:00" level=error msg="Handler for GET /v1.26/containers/foobar/json returned error: No such container: foobar"
Mar 23 10:41:14 admin1 dockerd[15092]: time="2017-03-23T10:41:14.722848795-05:00" level=error msg="Handler for GET /v1.26/images/foobar/json returned error: No such image: foobar"
Mar 23 10:41:14 admin1 dockerd[15092]: time="2017-03-23T10:41:14.724098031-05:00" level=error msg="Handler for GET /v1.26/networks/foobar returned error: network foobar not found"
Mar 23 10:41:14 admin1 dockerd[15092]: time="2017-03-23T10:41:14.724706840-05:00" level=error msg="Handler for GET /v1.26/volumes/foobar returned error: get foobar: no such volume"
Mar 23 10:41:14 admin1 dockerd[15092]: time="2017-03-23T10:41:14.745225668-05:00" level=error msg="Handler for GET /v1.26/plugins/foobar/json returned error: plugin \"foobar\" not found"
```

**Steps to reproduce the issue:**
1. Run `docker inspect foobar` (assuming you don't have a container named foobar)
2. Inspect syslogs (e.g. `journalctl --since "2 minutes ago" -u docker.service`)

**Additional information you deem important (e.g. issue happens only occasionally):**

OSSEC is commonly used IDS and alerts unrecognized syslog entries. Kontena and probably other scheduling systems may query non-existent containers frequently. This results in lots of error alerts for errors that really aren't important. If they are important to debug an issue then docker loglevel should be increased. The error alerting can be suppressed but these entries still are just wasting space and bandwidth considering lots of distributed systems transmit syslogs to central logging system.

**Output of `docker version`:**

```
Client:
 Version:      17.03.0-ce
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   60ccb22
 Built:        Thu Feb 23 11:02:43 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.0-ce
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   60ccb22
 Built:        Thu Feb 23 11:02:43 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by thaJeztah** [comment]
I'm not against changing them to "warning", at the same time I wonder why you have so many of these; I think there was a bug in cAdvisor polling the wrong container ID if you're using cAdvisor

ping @cpuguy83 wdyt?

**Comment by cpuguy83** [comment]
Probably not needed at all.
4xx responses are user error not system errors.

**Comment by colinmollenhour** [comment]
Yes, there is no doubt a bug in Kontena or cAdvisor somewhere causing the incorrect polling, but there will probably always be little things causing 404s here and there regardless and it just seems they do not need to be logged by default. Thanks for considering!

**Comment by colinmollenhour** [comment]
The "No such container" error should be a 404 response. There are two places where an error is logged in the format "Handler for ...":

[server.go:141](https://github.com/moby/moby/blob/2afb3efaa63537cc152437af3cd7a6743fcb3d8d/api/server/server.go#L141) - here it is only logged if it is a 5XX error.

[middleware.go:86](https://github.com/moby/moby/blob/8af4db6f002ac907b6ef8610b237879dfcaa5b7a/pkg/authorization/middleware.go#L86) - here it appears to always be logged so I suspect this is the source of the log messages.

If I'm correct, wouldn't it be consistent to update middleware.go to only log if the response is a 500 error also?

Thanks for your consideration.

**Comment by thaJeztah** [comment]
Are you interested in opening a pull request?

**Comment by cpuguy83** [comment]
I think likely we don't need to log in authz at all.

**Comment by colinmollenhour** [comment]
> Are you interested in opening a pull request?

Unfortunately I am not able to invest the time needed to open a PR myself since I do not have a dev environment for testing and don't know much about building large go projects and working with the test suite. For someone who knows what they are doing I was hoping this would be a simple fix.

**Comment by thaJeztah** [comment]
no problem, no worries!

@ripcurld0 @allencloud perhaps you're interested in looking at this?

**Comment by cameronthomas** [comment]
I am a beginner, but if @ripcurld0 or @allencloud are not able to work on it then I can work on it.

**Comment by thaJeztah** [comment]
Awesome, feel free to do so :+1:

**Comment by cameronthomas** [comment]
Ok, sounds good.

**Comment by cameronthomas** [comment]
@colinmollenhour, I’m trying to reproduce the error on my system. The output of of my `docker version` looks like:

```
Client:
 Version:      17.06.1-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:53:38 2017
 OS/Arch:      darwin/amd64
 
Server:
 Version:      17.06.1-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:54:55 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

You mentioned that Kontena was creating a log. If I don’t have Kontena up and running with Docker should expect any errors like the one you mentioned above in the log? You said you thought the logs were coming from [middleware.go:86](https://github.com/moby/moby/blob/8af4db6f002ac907b6ef8610b237879dfcaa5b7a/pkg/authorization/middleware.go#L86) If I just have docker, I am unable to find error messages like you mentioned above.

**Comment by colinmollenhour** [comment]
Doh, this does appear to be fixed somehow or another as of even 17.05.0-ce...

**Comment by cameronthomas** [comment]
Ok, I won't worry about it then.
